numpy/core/src/umath/ufunc_object.c

```diff
@@ -2864,7 +2864,16 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
     }
 
     /* Get the identity */
-    identity = _get_identity(ufunc, &reorderable);
+    if (initializer == NULL || initializer == Py_None) {
+        identity = _get_identity(ufunc, &reorderable);
+    } else {
+        /* Since initializer is passed through from the argument parsing,
+           we must increment the reference counter. */
+        identity = initializer;
+        Py_INCREF(initializer);
```

Comment:
This branch doesn't set `reorderable`, which is what is causing (intermittent) bugs.